src/cmap/wire_protocol/responses.ts

```diff
@@ -83,27 +95,98 @@ export class MongoDBResponse extends OnDemandDocument {
     return this.clusterTime ?? null;
   }
 
-  public override toObject(options: BSONSerializeOptions = {}): Record<string, any> {
+  public override toObject(options?: BSONSerializeOptions): Record<string, any> {
     const exactBSONOptions = {
-      useBigInt64: options.useBigInt64,
-      promoteLongs: options.promoteLongs,
-      promoteValues: options.promoteValues,
-      promoteBuffers: options.promoteBuffers,
-      bsonRegExp: options.bsonRegExp,
-      raw: options.raw ?? false,
-      fieldsAsRaw: options.fieldsAsRaw ?? {},
+      useBigInt64: options?.useBigInt64,
+      promoteLongs: options?.promoteLongs,
+      promoteValues: options?.promoteValues,
+      promoteBuffers: options?.promoteBuffers,
+      bsonRegExp: options?.bsonRegExp,
+      raw: options?.raw ?? false,
+      fieldsAsRaw: options?.fieldsAsRaw ?? {},
       validation: this.parseBsonSerializationOptions(options)
     };
     return super.toObject(exactBSONOptions);
   }
 
-  private parseBsonSerializationOptions({ enableUtf8Validation }: BSONSerializeOptions): {
+  private parseBsonSerializationOptions(options?: { enableUtf8Validation?: boolean }): {
     utf8: { writeErrors: false } | false;
   } {
+    const enableUtf8Validation = options?.enableUtf8Validation;
     if (enableUtf8Validation === false) {
       return { utf8: false };
     }
-
     return { utf8: { writeErrors: false } };
   }
 }
+
+/** @internal */
+export class CursorResponse extends MongoDBResponse {
+  static emptyGetMore = new CursorResponse(
+    Buffer.from(
+      'NgAAABBvawABAAAAA2N1cnNvcgAhAAAAEmlkAAAAAAAAAAAABG5leHRCYXRjaAAFAAAAAAAA',
+      'base64'
+    )
+  );
+
+  static override is(value: unknown): value is CursorResponse {
+    return value instanceof CursorResponse;
+  }
+
+  public id: Long | null = null;
+  public ns: MongoDBNamespace | null = null;
+  public batchSize = 0;
+
+  private batch: OnDemandDocument | null = null;
+  private values: Generator<OnDemandDocument, void, void> | null = null;
+  private iterated = 0;
+
+  constructor(b: Uint8Array, o?: number, a?: boolean) {
+    super(b, o, a);
+
+    if (this.isError) return;
```

Comment:
I don't like this. I would like to potentially write a helper that figures out before calling the responseType constructor if it is an error and then never calls the passed in responseType since we will always convert an ok:0 to a MongoServerError.

Reply:
Fixed